1.8/administration/overlay-networks/ip-per-container.md

```diff
@@ -194,6 +189,14 @@ The overlay networks install IPMASQ rules in order to allow containers to talk o
 
 To replace your overlay network, uninstall DC/OS and delete the replicated log on the master nodes and the iptable rules on the agent nodes. Then, reinstall with the desired networks specified in your `config.yaml` file.
 
+<a name="limitations"></a>
+# Limitations
+* In DC/OS overlay we slice the subnet of an overlay network into smaller subnets, and allocate these smaller subnets to agents. When an agent has exhausted its allocated address range and a framework tries to launch a container on the overlay network on this agent, the container launch will fail leading to a TASK_FAILED being reported to the framework. Since there is no API to report the exhaustion of addresses on an agent, it is up to the framework to infer that containers cannot be launched on an overlay network due to lack of IP addresses on the agent. This limitation of not being able to report address exhaustion has a direct impact on the behavior of frameworks, such as Marathon, that try to launch "services" with a specified number of instances. Due to this limitation, frameworks such as Marathon might not be able to complete their obligation of launching a service on an overlay network if they repeatedly try to launch instances of a service on an agent that has exhausted its allocated IP address range. The user should be aware of this limitation in order to debug issues using these frameworks when seeing task failures on an overlay network.
```

Comment:
How about:


* In DC/OS overlay we slice the subnet of an overlay network into smaller subnets and allocate these smaller subnets to agents. When an agent has exhausted its allocated address range and a framework tries to launch a container on the overlay network on this agent, the container launch will fail, leading to a TASK_FAILED message to the framework.

Since there is no API to report the exhaustion of addresses on an agent, it is up to the framework to infer that containers cannot be launched on an overlay network due to lack of IP addresses on the agent. This limitation has a direct impact on the behavior of frameworks, such as Marathon, that try to launch "services" with a specified number of instances. Due to this limitation, frameworks such as Marathon might not be able to complete their obligation of launching a service on an overlay network if they repeatedly try to launch instances of a service on an agent that has exhausted its allocated IP address range.

Keep this limitation in mind when debugging issues on frameworks that use an overlay network and see the `TASK_FAILED` message.

* The DC/OS overlay network does not allow frameworks to reserve IP addresses that result in ephemeral addresses for containers across multiple incarnations on the overlay network. 
 
To host services whose IP addresses are assumed to be static, administrators should use [VIPs (virtual IP addresses)](https://dcos.io/docs/1.7/usage/service-discovery/load-balancing/) to access their services.

* The DC/OS overlay network uses Linux bridge devices on agents to connect Mesos and Docker containers to the overlay network. The names of these bridge devices are derived from the overlay network name. Since Linux has a limitation of fifteen characters on network device names, there is a character limit of thirteen characters for the overlay network name (two characters are used to distinguish between a CNI bridge and a Docker bridge on the overlay network).

* Certain names are reserved and cannot be used as DC/OS overlay network names, primarily because DC/OS overlay uses Docker networking underneath to connect Docker containers to the overlay, which in turn reserves certain network names. The reserved names are: `host`, `bridge` and `default`."